### ISSUE TITLE
for loop and auto compiles

### ISSUE THREAD

**Issue by Indra7667** [glassmonkey/php-playground#232]
if user write a for loop like this
for(){
}
and then fill the () with the usual $i = int; $i< int; ...
the application will compiles the loop before $i++ is written, causing uncapped loop

**Comment by glassmonkey** [comment]
Yes, this is a known bug reported in the issue below. It causes the CPU to stack during an infinite loop.

https://github.com/glassmonkey/php-playground/issues/218

I believe we can improve this by using a service worker to separate the PHP process, but I haven't had the time to work on it yet. 

If you're interested, contributions are welcome!

**Comment by glassmonkey** [comment]
Thanks for the suggestion!

The original request was to add a flag to control synchronous vs asynchronous execution.
After looking into this, the core problem turned out to be CPU blocking on the main thread during PHP execution.

Instead of adding an execution-mode flag, PHP execution has now been fully moved into a Web Worker.
This isolates CPU-intensive work from the UI thread and keeps the playground responsive at all times.

With this change:
- The main thread is never blocked by PHP execution
- Execution is effectively non-blocking from the UI perspective
- An explicit sync/async flag is no longer necessary

I believe this addresses the underlying concern more robustly, so I’m marking this issue as resolved.
Thanks again for the feedback!
